microsoft/testsuites/xfstests/xfstesting.py

```diff
@@ -326,14 +451,24 @@ def _execute_xfstests(
                     "Current distro doesn't support btrfs file system."
                 )
 
-        _configure_disk(
-            node,
-            data_disk,
-            {test_dev: _test_folder, scratch_dev: _scratch_folder},
-            file_system=file_system,
-        )
+        if test_type != "cifs":
+            _configure_disk(
+                node,
+                data_disk,
+                {test_dev: _test_folder, scratch_dev: _scratch_folder},
+                file_system=file_system,
+            )
         xfstests = node.tools[Xfstests]
-        xfstests.set_local_config(scratch_dev, _scratch_folder, test_dev, _test_folder)
+        xfstests.set_local_config(
+            scratch_dev,
+            _scratch_folder,
+            test_dev,
+            _test_folder,
+            test_type,
+            _mount_opts,
+        )
+        if test_type == "cifs":
```

Comment:
Add comments, it's hard to understand why cifs will be changed to generic.

Reply:
update code